frame/support/src/traits/misc.rs

```diff
@@ -27,9 +27,39 @@ use sp_std::{cmp::Ordering, prelude::*};
 const DEFENSIVE_OP_PUBLIC_ERROR: &'static str = "a defensive failure has been triggered; please report the block number at https://github.com/paritytech/substrate/issues";
 const DEFENSIVE_OP_INTERNAL_ERROR: &'static str = "Defensive failure has been triggered!";
 
+macro_rules! defensive_without_err {
+	() => {
+		frame_support::log::error!(
+			target: "runtime",
+			"{}",
+			DEFENSIVE_OP_PUBLIC_ERROR
+		);
+		debug_assert!(false, "{}", DEFENSIVE_OP_INTERNAL_ERROR);
+	}
+}
+
+macro_rules! defensive_with_err {
```

Comment:
Feels like we can combine the two macros together with a proper naming, or in fact it can also eliminate the `defensive_path` function as well, if we just name this `defensive_path`, and give it a `()` arm.

Bonus points if you can use just one arm and do something like `($($err:expr)?)`.